content/blog/2022-05-04-e2e-cv-pt2.md

```diff
@@ -0,0 +1,197 @@
+can
+[persist](https://dvc.org/doc/user-guide/experiment-management/persisting-experiments)
+them by creating a branch out of an experiment:
+
+```bash
+$ dvc exp branch exp-04bcd my-branch
+Git branch 'my-branch' has been created from experiment 'exp-04bcd'.
+To switch to the new branch run:
+        git checkout my-branch
+```
+
+To track detailed information about the training process, we integrated
+[DVCLive](https://dvc.org/doc/dvclive) into the training code by
+[adding a callback object](https://github.com/iterative/magnetic-tiles-defect/blob/main/src/train_utils.py#L48)
+to the training function. DVCLive is a Python library for logging machine
+learning metrics and other metadata in simple file formats, which is fully
+compatible with DVC.
+
+## Collaboration and Reporting with Iterative Studio
+
+What if we needed to report the results to our team members or maybe hand over
+the project to one of them? How do we communicate everything we did since the
+conception of the project? What things resulted in the most significant
+improvements? What things didn't seem to matter at all?
+
+[Iterative Studio](https://studio.iterative.ai/) is a web-based application with
+seamless integration with DVC for data and model management, experiment
+tracking, visualization, and automation. It becomes especially valuable when
+collaborating with others on the same project or when there's a need to
+summarize the progress of the project through metrics and plots. All that's
+needed is to connect the project's repository with Studio. Then Studio will
+automatically parse all required information from `dvc.yaml`, `params.yaml`, and
+other text files that DVC recognizes. The result will be a repository view. The
+view for our project is
+[here](https://studio.iterative.ai/user/alex000kim/views/magnetic-tiles-defect-5kozhnu9jo).
+It displays commits, metrics, parameters, the remote location of data and models
+tracked by DVC, and more.
+
+In the screenshot below, you can see that we created a separate `exp` branch
+that displays the results of the local experiments that we decided to upload to
+our remote repository, like trying different learning rates and batch sizes.
+Note that earlier, we discarded all local experiments whose performance we
+weren't satisfied with.
+
+![Studio view](/uploads/images/2022-05-04/studio_view.png '=800')
+
+Below we can see the evolution of the key metrics and the value of the loss
+function throughout training (enabled by the earlier integration of
+[DVCLive](https://dvc.org/doc/dvclive)) for a set of selected commits.
+
+![DVCLive metrics displayed in
+Studio](/uploads/images/2022-05-04/dvc_live_studio.png '=800')
+
+Now, for example, if we see that the loss function hasn't reached a plateau
+after a certain number of epochs, we'll try increasing this number. Or, even
+worse, if we see the loss function growing over time, it'll be an indication
+that our learning rate may be too high.
```

Comment:
I feel there needs to be a concluding sentence in this paragraph that leads back to the point of this post like, "So it will be back to the experimentation drawing board." or something like that.  Feels abrupt before jumping to the summary as is.

Reply:
Yup, added a concluding sentence.